modules/memory-management/src/main/java/org/mule/runtime/internal/memory/management/DefaultMemoryManagementService.java

```diff
@@ -26,17 +39,50 @@ public static DefaultMemoryManagementService getInstance() {
 
   @Override
   public void dispose() {
-    // TODO EE-8096 Default ByteBufferProvider implementation in memory management
+    byteBufferProviders.values().forEach(ByteBufferProvider::dispose);
+    byteBufferProviders.clear();
   }
 
   @Override
   public void initialise() throws InitialisationException {
-    // TODO EE-8096 Default ByteBufferProvider implementation in memory management
+    // Nothing to do.
+  }
+
+  @Override
+  public ByteBufferProvider<ByteBuffer> getByteBufferProvider(String name, ByteBufferType byteBufferType, int maxBufferSize,
+                                                              ByteBufferPoolConfiguration poolConfiguration) {
+    if (byteBufferProviders.containsKey(name)) {
+      throw new IllegalArgumentException(format(DUPLICATE_BYTE_BUFFER_PROVIDER_NAME, name));
+    }
+
+    ByteBufferProvider<ByteBuffer> byteBufferProvider = buildByteBufferProviderFrom(byteBufferType)
+        .withMaxBufferSize(maxBufferSize)
+        .withPoolConfiguration(poolConfiguration)
+        .build();
+
+    byteBufferProviders.put(name, byteBufferProvider);
+
+    return byteBufferProvider;
   }
 
   @Override
-  public ByteBufferProvider getByteBufferProvider(String name, ByteBufferProviderConfiguration configuration) {
-    // TODO EE-8096 Default ByteBufferProvider implementation in memory management
-    return null;
+  public ByteBufferProvider<ByteBuffer> getByteBufferProvider(String name, ByteBufferType byteBufferType) {
+    if (byteBufferProviders.containsKey(name)) {
+      throw new IllegalArgumentException(format(DUPLICATE_BYTE_BUFFER_PROVIDER_NAME, name));
+    }
+
+    ByteBufferProvider<ByteBuffer> byteBufferProvider = buildByteBufferProviderFrom(byteBufferType)
+        .build();
+
+    byteBufferProviders.put(name, byteBufferProvider);
+
+    return byteBufferProvider;
   }
+
+  @Override
+  public void disposeByteBufferProvider(String name) {
+    byteBufferProviders.get(name).dispose();
+    byteBufferProviders.remove(name);
```

Comment:
This would avoid double-access to the map:
byteBufferProviders.remove(name).dispose();
Also, since it's a ConcurrentHashMap I would assume it is accessed from different threads. These methods should be thread-safe